core-api/src/main/java/com/optimizely/ab/internal/ExperimentUtils.java

```diff
@@ -73,14 +73,15 @@ public static boolean isUserInExperiment(@Nonnull ProjectConfig projectConfig,
         }
     }
 
-    public static @Nullable
-    Boolean evaluateAudience(@Nonnull ProjectConfig projectConfig,
+    @Nullable
+    public static Boolean evaluateAudience(@Nonnull ProjectConfig projectConfig,
                              @Nonnull Experiment experiment,
                              @Nonnull Map<String, ?> attributes) {
         List<String> experimentAudienceIds = experiment.getAudienceIds();
 
         // if there are no audiences, ALL users should be part of the experiment
         if (experimentAudienceIds.isEmpty()) {
+            logger.debug("There is no Audience associated with experiment {}", experiment.getKey());
```

Comment:
Good thinking here. However, I have a couple thoughts

* We should log the `audienceIds` that we use even if (_especially_ if) they're non-empty
* Perhaps we should use the same type of log message that we use for `audienceConditions`:
  ```
  "Evaluating audiences for experiment \"{}\": \"{}\""
  ```
  Example log message:
  ```
  Evaluating audiences for experiment "123": ["456", "789"]
  ```

Reasoning:
* This makes the `audienceIds` -> `audienceConditions` migration more transparent to the customer.
* This may also increase the consistency of our log messages across SDKs. The JavaScript SDK and other dynamic-language SDKs use the same evaluation logic for `audienceIds` and `audienceConditions`.